File: svws-webclient/core/src/asd/validate/klassen/ValidatorKob00KlassenOrganisationsformBerufsbildend.ts
import type { Supplier } from '../../../java/util/function/Supplier';
import { ValidatorKob01KlassenOrganisationsformBerufsbildend } from '../../../asd/validate/klassen/ValidatorKob01KlassenOrganisationsformBerufsbildend';
import { Class } from '../../../java/lang/Class';
import { ValidatorKontext } from '../../../asd/validate/ValidatorKontext';
import { Validator } from '../../../asd/validate/Validator';

export class ValidatorKob00KlassenOrganisationsformBerufsbildend extends Validator {

	/**
	 * Orgaform
	 */
	private readonly _idOrgaform: Supplier<number | null>;

	private static readonly FEHLERTEXT: string = "Organisationsform der Klasse: Kein Wert vorhanden.";


	/**
	 * Erstellt einen neuen Validator mit den übergebenen Daten und dem übergebenen Kontext
	 *
	 * @param idOrgaform  die Organisationsform
	 * @param kontext     der Kontext des Validators
	 */
	public constructor(idOrgaform: Supplier<number | null>, kontext: ValidatorKontext) {
		super(kontext);
		this._idOrgaform = idOrgaform;
		this._validatoren.add(new ValidatorKob01KlassenOrganisationsformBerufsbildend(this.getNotNullSupplierLong(idOrgaform), kontext));
	}

	protected pruefe(): boolean {
		const idOrgaform: number | null = this._idOrgaform.get();
		if (idOrgaform === null) {
			this.addFehler(0, ValidatorKob00KlassenOrganisationsformBerufsbildend.FEHLERTEXT);
			return false;
		}
		return true;
	}

	transpilerCanonicalName(): string {
		return 'de.svws_nrw.asd.validate.klassen.ValidatorKob00KlassenOrganisationsformBerufsbildend';
	}

	isTranspiledInstanceOf(name: string): boolean {
		return ['de.svws_nrw.asd.validate.klassen.ValidatorKob00KlassenOrganisationsformBerufsbildend', 'de.svws_nrw.asd.validate.BasicValidator', 'de.svws_nrw.asd.validate.Validator'].includes(name);
	}

	public static readonly class = new Class<ValidatorKob00KlassenOrganisationsformBerufsbildend>('de.svws_nrw.asd.validate.klassen.ValidatorKob00KlassenOrganisationsformBerufsbildend');

}

export function cast_de_svws_nrw_asd_validate_klassen_ValidatorKob00KlassenOrganisationsformBerufsbildend(obj: unknown): ValidatorKob00KlassenOrganisationsformBerufsbildend {
	return obj as ValidatorKob00KlassenOrganisationsformBerufsbildend;
}

File: svws-asd/src/main/java/de/svws_nrw/asd/validate/klassen/ValidatorKob00KlassenOrganisationsformBerufsbildend.java
package de.svws_nrw.asd.validate.klassen;

import java.util.function.Supplier;

import de.svws_nrw.asd.validate.Validator;
import de.svws_nrw.asd.validate.ValidatorKontext;
import de.svws_nrw.transpiler.annotations.AllowNull;
import jakarta.validation.constraints.NotNull;

/**
 * Dieser Validator führt eine Statistikprüfung auf die Orgaform Berufsbildend bei den Stammdaten
 * einer Klasse aus.
 */
public final class ValidatorKob00KlassenOrganisationsformBerufsbildend extends Validator {


		/** Orgaform */
		private final @NotNull Supplier<@AllowNull Long> _idOrgaform;
		private static final @NotNull String FEHLERTEXT = "Organisationsform der Klasse: Kein Wert vorhanden.";

		/**
		 * Erstellt einen neuen Validator mit den übergebenen Daten und dem übergebenen Kontext
		 *
		 * @param idOrgaform  die Organisationsform
		 * @param kontext     der Kontext des Validators
		 */
		public ValidatorKob00KlassenOrganisationsformBerufsbildend(
				final @NotNull Supplier<@AllowNull Long> idOrgaform,
				final @NotNull ValidatorKontext kontext) {
			super(kontext);
			_idOrgaform = idOrgaform;

			_validatoren.add(
					new ValidatorKob01KlassenOrganisationsformBerufsbildend(getNotNullSupplierLong(idOrgaform), kontext));
		}


		@Override
		protected boolean pruefe() {
			final @AllowNull Long idOrgaform = _idOrgaform.get();

				if (idOrgaform == null) {
					addFehler(0, FEHLERTEXT);
					return false;
				}
			return true;
		}
	}
